core/src/main/java/io/aiven/klaw/service/UiControllerLoginService.java

```diff
@@ -207,18 +203,21 @@ private static void sendResponse(
   }
 
   private Pair<Boolean, String> getRoleFromTokenAuthorities(
-      DefaultOAuth2User defaultOAuth2User,
+      Collection<? extends GrantedAuthority> authorities,
       String userName,
       Set<String> klawRoles,
       HttpServletResponse response) {
     String roleFromClaim = null;
-    Collection<? extends GrantedAuthority> authorities = defaultOAuth2User.getAuthorities();
     int claimMatched = 0;
     for (GrantedAuthority authority : authorities) {
-      String authorityRole = authority.getAuthority().substring(5); // ex : ROLE_USER
-      if (klawRoles.stream().anyMatch(authorityRole::equalsIgnoreCase)) {
-        claimMatched++;
-        roleFromClaim = authorityRole;
+      String authorityStr = authority.getAuthority();
+      int indexOfRoleSeparator = authorityStr.indexOf("_");
+      if (indexOfRoleSeparator > 0) {
+        String authorityRole = authorityStr.substring(indexOfRoleSeparator + 1); // ex : ROLE_USER
+        if (klawRoles.stream().anyMatch(authorityRole::equalsIgnoreCase)) {
```

Comment:
Looks like it does not use feature of sets... 
Currently in worst case it iterates over sets instead of checking by `get`.
Could we store in klawRoles only lower case or only uppercase?
if yes than it could be speed up by replacing iteration with get

Reply:
these roles are usually not more than 5 or 10. but i get your point. may be we can apply on topic/connector sets